clash-prelude/src/Clash/Explicit/Signal.hs

```diff
@@ -859,4 +861,88 @@ sampleWithResetN
 sampleWithResetN nReset nSamples f =
   take nSamples (sampleWithReset nReset f)
 
+-- | Simulate a component until it matches a condition
+--
+-- It prints a message of the form
+--
+-- "Signal sampled for N cycles until value X"
```

Comment:
I wonder if this should be in a `@` block and without quotation marks, so it's clearer at a glance that this is console output